blocks/filter.py

```diff
@@ -96,6 +96,16 @@ class object contained in the list.
     """
     def __init__(self, roles=None, bricks=None, each_role=False, name=None,
                  name_regex=None, applications=None):
+        if bricks is not None and not isinstance(bricks, Brick):
```

Comment:
IMHO we should just get rid of this convention that both `Brick` and a list of bricks can be passed. It doesn't make sense to call the argument `bricks` and then allow a single one.

Anyway, this is probably shorter (and likewise for `applications`):
```python
if isinstance(bricks, Brick):
   bricks = [bricks]
if not all(isinstance(brick, Brick) for brick in bricks):
    raise ValueError("`bricks` should be a list of bricks")
```
Note that you should be raising a `ValueError`, not use `assert`. Assertions are predicates that should *always* be true. They're used to catch bugs in the code, not to alert users of wrong inputs. See [the documentation](http://blocks.readthedocs.org/en/latest/development/index.html#validating-function-arguments).